npm-package/src/controllers/otelController.ts

```diff
@@ -1,25 +1,95 @@
-import { Request, Response, NextFunction, RequestHandler } from 'express';
+import { Request, Response, NextFunction } from 'express'
 
-interface OtelController {
-    parseAllRequest: RequestHandler,
-    parseNodeRequest: RequestHandler,
-    parseNextJSRequest: RequestHandler
-}
-
-export const otelController: OtelController = {
-    parseAllRequest : (req: Request, res: Response, next: NextFunction) => {
-        //need to diffrentiate between nextJS and Node application since parsing of the data is different
-
-    },
-    parseNextJSRequest : (req: Request, res: Response, next: NextFunction) => {
+type MainController = (req: Request, res: Response, next: NextFunction) => void
+type HelperController = (req: Request) => {}
 
-    },
-    parseNodeRequest : (req: Request, res: Response, next: NextFunction) => {
+export interface OtelControllerType {
+  parseAllRequest: MainController
+  parseNodeRequest: HelperController
+  parseNextJSRequest: HelperController
+}
 
-    } 
-};
+export const otelController: OtelControllerType = {
+  parseAllRequest: (req: Request, res: Response, next: NextFunction) => {
+    //need to diffrentiate between nextJS and Node application since parsing of the data is different
+    let data = {}
+
+    if (
+      req.body.resourceSpans[0].resource.attributes[4].value.stringValue ===
+      'next.js'
+    ) {
+      data = otelController.parseNextJSRequest(req)
+    }
+
+    console.log(data)
+
+    return next()
+  },
+  parseNextJSRequest: (req: Request) => {
+    const data: any = {}
+
+    if (req.body.resourceSpans[0].scopeSpans[0].spans[0]?.traceId) {
+      data.traceId = req.body.resourceSpans[0].scopeSpans[0].spans[0]?.traceId
+    }
+
+    if (req.body.resourceSpans[0].scopeSpans[0].spans[0]?.spanId) {
+      data.spanId = req.body.resourceSpans[0].scopeSpans[0].spans[0]?.spanId
+    }
+
+    if (
+      req.body.resourceSpans[0].resource.attributes[4].value.stringValue ===
+      'next.js'
+    ) {
+      const applicationType =
+        req.body.resourceSpans[0].resource.attributes[4].value.stringValue
+      data.ApplicationType = applicationType
+    }
+
+    if (req.body.resourceSpans[0].resource?.attributes[0]?.value?.stringValue) {
+      data.originatingService =
+        req.body.resourceSpans[0].resource?.attributes[0]?.value?.stringValue
+    }
+
+    if (req.body.resourceSpans[0]?.scopeSpans[0]?.spans[0]?.name) {
+      data.method = req.body.resourceSpans[0]?.scopeSpans[0]?.spans[0]?.name
+    }
+
```

Comment:
we might be able to convert this series of if/then statement into case/switches

Reply:
I was thinking about different formats for parsing the data, and I came to the conclusion that accessing the data directly would be the most efficient; however, we would need the if statements to ensure data sanitation. However, we can look into the cases/switch functionality as a team during class time.